tb_plugin/fe/src/app.tsx

```diff
@@ -153,8 +162,10 @@ export const App = () => {
       try {
         const runs = await api.defaultApi.runsGet()
         setRuns(runs)
+        setHasRuns(true)
```

Comment:
Won't this set `hasRuns` to `true` even if the runs returned from the backend is empty?

Reply:
I don't think you need a new state variable.  Can you just check `if (runs)` in any place you would do `if (hasRuns)`?

Reply:
I need the hasRun to differentiate whether the load is done but no any run or the loading is in progress

Reply:
Ah, I see.  The backend returns a 404, not an empty list when it finishes scanning the directory if there are no runs.